src/gtpu/encap.c

```diff
@@ -830,18 +847,21 @@ static int gtp5g_fwd_skb_encap(struct sk_buff *skb, struct net_device *dev,
 
     pdr->ul_pkt_cnt++;
     pdr->ul_byte_cnt += skb->len; /* length without GTP header */
-    GTP5G_INF(NULL, "PDR (%u) UL_PKT_CNT (%llu) UL_BYTE_CNT (%llu)", pdr->id, pdr->ul_pkt_cnt, pdr->ul_byte_cnt);
-
+    GTP5G_INF(NULL, "PDR (%u) UL_PKT_CNT (%llu) UL_BYTE_CNT (%llu)", pdr->id, pdr->ul_pkt_cnt, pdr->ul_byte_cnt);    
+ 
+    if (pdr->urr_num != 0) {
+        if (check_urr(pdr, far, volume, volume_mbqe, true, drop_pkt) < 0)
+            GTP5G_ERR(pdr->dev, "Fail to send Usage Report");
+    }
+    
+    if (color == Red){
+        return -1;
```

Comment:
There are more than one semantics for using -1 as return values in this function. That would be are to debug especially when the errnos are returned when no error messages are printed.